pkg/tags/tags.go

```diff
@@ -0,0 +1,100 @@
+// Copyright 2018-2022 CERN
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// In applying this license, CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+
+package tags
+
+import "strings"
+
+var (
+	// character to separate tags
+	_tagsep = ","
+	// maximum number of tags
+	_maxtags = 100
+)
+
+// Tags is a helper struct for merging, deleting and deduplicating the tags while preserving the order
+type Tags struct {
+	sep     string
+	maxtags int
+
+	t       []string
+	exists  map[string]bool
+	numtags int
+}
+
+// FromString creates a Tags struct from a string
+func FromString(s string) *Tags {
+	t := &Tags{sep: _tagsep, maxtags: _maxtags, exists: make(map[string]bool)}
+
+	tags := strings.Split(s, t.sep)
+	for _, tag := range tags {
+		t.t = append(t.t, tag)
+		t.exists[tag] = true
+		t.numtags++
+	}
+	return t
+}
+
+// AddString appends the the new tags and returns true if at least one was appended
+func (t *Tags) AddString(s string) bool {
```

Comment:
is something else planned than string? why not just `Add`?

Reply:
If I would call `tags.Add(s)` I would expect `s` being one single tag. This is different here. `s` is a comma seperated list of tags. Maybe `AddMultiple`?

Reply:
YES :)

Reply:
Did go with `AddList` `RemoveList` `FromList` because thats what it does. It adds a list of tags